Description: _use_system_libnpx_1
--- a/Makefile
+++ b/Makefile
@@ -9,8 +9,7 @@
 cli_mandocs = $(shell find docs/content/cli-commands -name '*.md' \
                |sed 's|.md|.1|g' \
                |sed 's|docs/content/cli-commands/|man/man1/|g' ) \
-               man/man1/npm-README.1 \
-               man/man1/npx.1
+               man/man1/npm-README.1
 
 files_mandocs = $(shell find docs/content/configuring-npm -name '*.md' \
                |sed 's|.md|.5|g' \
@@ -84,7 +83,7 @@
 	@[ -d man/man1 ] || mkdir -p man/man1
 	node scripts/docs-build.js $< $@
 
-man/man1/npx.1: node_modules/libnpx/libnpx.1
+man/man1/npx.1: /usr/lib/nodejs/libnpx/libnpx.1
 	cat $< | sed s/libnpx/npx/ > $@
 
 man/man5/npm-json.5: man/man5/package.json.5
